business/src/main/java/org/mskcc/cbio/portal/service/ApiService.java

```diff
@@ -74,6 +80,165 @@
 		return cancerTypeMapper.getCancerTypes(cancer_type_ids);
 	}
 
+        @Transactional
+	public List<DBAltCount> getMutations(String type, Boolean per_study, List<String> ids,  List<String> genes, List<Integer> starts, List<Integer> ends, List<String> echo) {
```

Comment:
The code here should not assume there are IDs. The user may not provide an ID. They can provide anything in the URL to echo back.